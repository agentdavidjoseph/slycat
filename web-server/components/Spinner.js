import React from "react";

class Spinner extends React.Component {
  render() {
    return (
    	<div className="d-flex justify-content-center mt-4">
        <div className="spinner-border" role="status">
          <span className="sr-only">Loading...</span>
        </div>
      </div>
    );
  }
}

export { Spinner };
