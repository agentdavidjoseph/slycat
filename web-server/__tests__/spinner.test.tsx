
import React from 'react';
import Spinner, 
{SpinnerProps, SpinnerState} 
from "components/Spinner";
import { mount } from "enzyme";


describe('when loading a bad marking the UnrecognizedMarkingWarning',() =>{ 
  const render = mount(
    <Spinner/>
  );
  test('we have no props on initial load', () => {
    const props = render.props() as SpinnerProps;
    expect(props).toMatchSnapshot();
  });

  test('we no state', () => {
    const state = render.state() as SpinnerState;
    expect(state).toMatchSnapshot();
  });

  test('we should have a full initialization', () => {
    expect(render).toMatchSnapshot();
  });

});

export default undefined