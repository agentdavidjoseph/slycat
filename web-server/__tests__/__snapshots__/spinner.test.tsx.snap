// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading a bad marking the UnrecognizedMarkingWarning we have no props on initial load 1`] = `Object {}`;

exports[`when loading a bad marking the UnrecognizedMarkingWarning we no state 1`] = `Object {}`;

exports[`when loading a bad marking the UnrecognizedMarkingWarning we should have a full initialization 1`] = `
<Spinner>
  <div
    className="d-flex justify-content-center mt-4"
  >
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</Spinner>
`;
